tests/libstorage/datavolume.go

```diff
@@ -80,36 +77,7 @@ func EventuallyDV(dv *v1beta1.DataVolume, timeoutSec int, matcher gomegatypes.Go
 }
 
 func EventuallyDVWith(namespace, name string, timeoutSec int, matcher gomegatypes.GomegaMatcher) {
-	virtCli := kubevirt.Client()
-
-	if !IsDataVolumeGC(virtCli) {
-		Eventually(ThisDVWith(namespace, name), timeoutSec, time.Second).Should(matcher)
-		return
-	}
-
-	ginkgo.By("Verifying DataVolume garbage collection")
-	var dv *v1beta1.DataVolume
-	Eventually(func() *v1beta1.DataVolume {
-		var err error
-		dv, err = ThisDVWith(namespace, name)()
-		Expect(err).ToNot(HaveOccurred())
-		return dv
-	}, timeoutSec, time.Second).Should(Or(BeNil(), matcher))
-
-	if dv != nil {
-		if dv.Status.Phase != v1beta1.Succeeded {
-			return
-		}
-		if dv.Annotations["cdi.kubevirt.io/storage.deleteAfterCompletion"] == "true" {
-			Eventually(ThisDV(dv), timeoutSec).Should(BeNil())
-		}
-	}
-
-	Eventually(func() bool {
-		pvc, err := ThisPVCWith(namespace, name)()
-		Expect(err).ToNot(HaveOccurred())
-		return pvc != nil && pvc.Spec.VolumeName != ""
-	}, timeoutSec, time.Second).Should(BeTrue())
+	Eventually(ThisDVWith(namespace, name), timeoutSec, time.Second).Should(matcher)
 }
 
 func DeleteDataVolume(dv **v1beta1.DataVolume) {
```

Comment:
Do we really need this? I can't remember why we had to dive into the **v1beta1.DataVolume realm

Reply:
We use `**v1beta1.DataVolume` to set `Context()`-shared dv to nil after we `Delete()` it in the `AfterEach()`, so we won't carry over the dv value of previous test, and we know if it was created/deleted. Removing it will add that same code to some of the `AfterEach`s.

Reply:
Regarding the `EventuallyDV/With()`, I think it's preferable as it's used all over the storage tests and shorter and more readable than the explicit call.

Reply:
The DVs are deleted anyway as part of the global `testCleanup` after each test, so we don't have to delete explicitly.
If we have to nil out some global variables, we can do that in the AfterEach blocks which I believe is what we usually do

Reply:
Don't need to nil them out? I guess not if the tests pass?

Reply:
Going over the tests on the last commit, it seems we have no reason to nil them.

Reply:
Great